### PR TITLE
Use native pickling instead of JSON for zmq interface

### PR DESCRIPTION
**Pull request by slacrherbst**
This change uses native python pickling instead of json encoding for sending data between the zmq client and server. 

This also uses raw bytes for data transfer between the client and server instead of passing and decoding strings.

The C++ raw client interface is still supported on a third server port.

This has resulted in a 50% speedup of zmq based transactions which should make the GUI more responsive. The Rogue zmq remains faster than the epics3 interface:

For a list transfer:
   Epics list caget Rate = 1132.6 hz
   Epics list caput Rate = 1218.6 hz
   Epics list pv.get Rate = 1145.9 hz
   Epics list pv.put Rate = 1261.5 hz
   zmq list get Rate = 3159.4 hz
   zmq list set Rate = 3048.9 hz

For a single value transfer:
   Epics caget Rate = 1668.3 hz
   Epics caput Rate = 2379.5 hz
   Epics pv.get Rate = 2105.4 hz
   Epics pv.put Rate = 2457.7 hz
   zmq get Rate = 4318.7 hz
   zmq set Rate = 4028.2 hz

### CODE REVIEW

**Comment by bengineerd** [comment]
I assume this doesn't break backward compatibility of the ZMQ protocol?

**Comment by slacrherbst** [comment]
It does. The client and server versions will have to match.

**Comment by slacrherbst** [comment]
This PR is now ready and the old raw CPP interface has been restored.

File: tests/epics_client_testing.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
+
```
Comment (ruck314) [inline tests/epics_client_testing.py:1]: missing SLAC licensing header

**Pull request merged**